pkgs/development/libraries/protobuf/generic.nix

```diff
@@ -36,10 +36,19 @@ stdenv.mkDerivation (finalAttrs: {
     inherit hash;
   };
 
-  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
-    substituteInPlace src/google/protobuf/testing/googletest.cc \
-      --replace 'tmpnam(b)' '"'$TMPDIR'/foo"'
-  '';
+  postPatch =
+    lib.optionalString (stdenv.hostPlatform.isDarwin && lib.versionOlder version "29") ''
+      substituteInPlace src/google/protobuf/testing/googletest.cc \
+        --replace-fail 'tmpnam(b)' '"'$TMPDIR'/foo"'
+    ''
+    + (lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+      substituteInPlace cmake/protobuf-config.cmake.in \
+        --replace-quiet 'COMMAND protobuf::protoc' \
+        'COMMAND ${lib.getExe buildPackages."protobuf_${lib.versions.major version}"}'
+      substituteInPlace cmake/protobuf-generate.cmake \
```

Comment:
There's no such file when building `pkgsCross.aarch64-multiplatform.protobuf_21`
```
unpacking source archive /nix/store/2nz7rsndid4iwl7dxfzgwm2xjzwr4y1i-source
source root is source
Running phase: patchPhase
applying patch /nix/store/0sgpq8qlrcgjsvxjrk3p4q2nxn5km2c5-a7324f88e92bc16b57f3683403b6c993bf68070b.patch
patching file cmake/install.cmake
substitute(): ERROR: file 'cmake/protobuf-generate.cmake' does not exist
```

Reply:
Oh, I must have missed that version. Will check again.

Reply:
Fixed. Build is still running but the patchPhase passed.